internal/loader/configuration_setting_loader.go

```diff
@@ -71,13 +67,24 @@ type ConfigurationSettingsRetriever interface {
 	ResolveSecretReferences(ctx context.Context, kvReferencesToResolve map[string]*TargetSecretReference, kvResolver SecretReferenceResolver) (map[string]corev1.Secret, error)
 }
 
-type GetSettingsFunc func(ctx context.Context, filters []acpv1.Selector, client *azappconfig.Client, c chan []azappconfig.Setting, e chan error)
-
 type ServicePrincipleAuthenticationParameters struct {
 	ClientId     string
 	ClientSecret string
 	TenantId     string
 }
+type settingsGetter struct {
```

Comment:
How about put `settingsGetter`,`sentinelGetter` and their implementations into a separate file